lambdas/journeyengine/src/main/java/uk/gov/di/ipv/core/journeyengine/JourneyEngineHandler.java

```diff
@@ -147,33 +140,36 @@ private JourneyEngineResult executeJourneyEvent(
                             new JourneyResponse(criStartUri + UK_PASSPORT_CRI_ID));
                     break;
                 case CRI_UK_PASSPORT:
-                    if (journeyStep.equals(NEXT_STEP)) {
+                    if (journeyStep.equals(JourneyStep.NEXT.toString())) {
                         updateUserState(CRI_ADDRESS, ipvSessionItem);
                         builder.setJourneyResponse(
                                 new JourneyResponse(criStartUri + ADDRESS_CRI_ID));
-                    } else if (journeyStep.equals(ERROR_STEP)) {
+                    } else if (journeyStep.equals(JourneyStep.ERROR.toString())) {
+                        updateUserState(CRI_ERROR, ipvSessionItem);
+                        builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
+                    } else if (journeyStep.equals(JourneyStep.FAIL.toString())) {
                         updateUserState(CRI_ERROR, ipvSessionItem);
                         builder.setPageResponse(new PageResponse(PYI_TECHNICAL_ERROR_PAGE.value));
```

Comment:
In the fail case for passport, i think you want to be redirecting to the specific error page, rather than the technical error page. It was added in [this PR](https://github.com/alphagov/di-ipv-core-front/pull/191).

Reply:
Done